### PR TITLE
datetime formats preservation and inference

### PR DESCRIPTION
**Pull request by izulin**
### Context

Formatting info should be preserved for datetime values. Formatting should be used for inferrence of formatting under arithmetic operations.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #626 
2.
3.

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation,
- [x] I described the modification in the CHANGELOG.md file.

### CODE REVIEW

File: src/CellContentParser.ts [hunk 1]
```diff
@@ -127,11 +134,11 @@ export class CellContentParser {
     } else if (typeof content === 'boolean') {
       return new CellContent.Boolean(content)
     } else if (content instanceof Date) {
-      return new CellContent.Number(this.dateHelper.dateToNumber({
+      return new CellContent.Number(new DateNumber(this.dateHelper.dateToNumber({
         day: content.getDate(),
         month: content.getMonth() + 1,
         year: content.getFullYear()
-      }))
+      }), "Date()"))
```
Comment (izulin) [inline src/CellContentParser.ts]: @wojciechczerniak not really sure how to solve it (what is the format of Date that came from Date() JS object)?

File: test/serialization.spec.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ describe('serialization', () => {
     expect(engine1.getCellValueDetailedType(adr('F1'))).toEqual(CellValueDetailedType.NUMBER_CURRENCY)
 
     expect(engine1.getCellSerialized(adr('G1'))).toEqual('12/01/15')
-    expect(engine1.getCellValueFormat(adr('G1'))).toEqual(undefined)  //this is wrong, see issue 626
+    expect(engine1.getCellValueFormat(adr('G1'))).toEqual('DD/MM/YY')
```
Comment (wojciechczerniak) [inline test/serialization.spec.ts:36]: I've seen that you handled native `Date()` object. Will it work with serialization? Can we add an example to the dataset in this test?

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]
All great 👏  just one little request for a test

File: src/CellContentParser.ts (hunk 1 above)
Comment (wojciechczerniak) [reply src/CellContentParser.ts]: I didn't noticed the comment, sorry. I think that if it was a Date object, we may serialize to Date object as well.

But I'm not sure how to handle the logic `= Date() + DD/MM/YY` 🤔

Comment (izulin) [reply src/CellContentParser.ts]: The left operand aways wins in our logic and then its format is preserved.

**Review by wojciechczerniak — APPROVED** [review]
👍

**Pull request merged**